src/omero_upload/library.py

```diff
@@ -56,7 +56,15 @@ def upload_ln_s(client, filepath, omero_data_dir, mimetype=None):
         omero_data_dir, 'Files', omero.util.long_to_path(fo.id))
 
     log.debug('OriginalFile:%d Deleting %s', fo.id, omero_path)
-    os.remove(omero_path)
+    try:
+        os.remove(omero_path)
+    except OSError:
+        log.error(
+            'Unable to delete, do you have direct access to the OMERO '
```

Comment:
Assuming user is on server A but logged into server B this has the potential of deleting unintended files. See https://github.com/ome/omero-blitz/blob/master/src/main/java/ome/formats/importer/transfers/AbstractExecFileTransfer.java#L105 for how inplace import handles this. If we had this `checkLocation` location in a re-usable location, then the upcoming import_library could re-use it.

Reply:
I have wondered if we can compare the repo `.lock` file parent directory's name with something we can get from the Blitz API so as to be really sure.

Reply:
This is partly related to https://github.com/ome/omero-upload/pull/3/files#r261178274 which had been marked as resolved.

Re-using the `checkLocation` logic https://github.com/ome/omero-upload/pull/3/files#r269081424, it might be sufficient to update the logic above to:

1- create a placeholder file containing a UUID instead of an empty string
2- check the file exists and contains the UUID i.e. make sure we are on the right server
3- remove the file